File: block_sites/go/block.go
package main

import (
	"io/ioutil"
	"os"
	"regexp"
	"strings"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func block() {
	dat, err := ioutil.ReadFile("./testhosts")
	check(err)

	fileStr := string(dat)

	lines := strings.Split(fileStr, "\n")

	blockedSitesSection := false

	var nextLines []string

	for _, line := range lines {
		line = strings.Trim(line, " ")
		if regexp.MustCompile(`blocked\ssites\sstart`).MatchString(line) {
			blockedSitesSection = true
		}

		if regexp.MustCompile(`blocked\ssites\send`).MatchString(line) {
			blockedSitesSection = false
		}

		if blockedSitesSection && regexp.MustCompile(`^\s*#+\s*127\.0\.0\.1`).MatchString(line) {
			line = regexp.MustCompile(`^#+\s*`).ReplaceAllString(line, "")
		}

		nextLines = append(nextLines, line)
	}

	f, err := os.Create("./testhosts")
	check(err)

	defer f.Close()

	_, err = f.WriteString(strings.Join(nextLines, "\n"))

	if err != nil {
		return
	}
}

func main() {
	block()
}

File: block_sites/deno/block.ts
const file = await Deno.readTextFile('./testhosts');

let blocked_sites_section = false;

const lines = file.split('\n').map((line): string => {
  if (/blocked\ssites\sstart/.test(line)) {
    blocked_sites_section = true;
  } else if (/blocked\ssites\send/.test(line)) {
    blocked_sites_section = false;
  }

  if (blocked_sites_section && /^\s*#+\s*127\.0\.0\.1/.test(line)) {
    line = line.replace(/^\s*#+\s*/, '');
  }

  return line;
});

await Deno.writeTextFile('./testhosts', lines.join('\n'));
